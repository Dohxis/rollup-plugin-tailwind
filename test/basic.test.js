const tailwind = require("../pkg/dist-node/index").default;
const { rollup } = require("rollup");

const generateCode = async file => {
  const bundle = await rollup({
    input: file,
    plugins: [
      tailwind({
        config: "test/samples/tailwind.config.js",
        only: ["*.js"]
      })
    ]
  });
  const { output } = await bundle.generate({ format: "es" });
  return output[0].code;
};

describe("transform tailwind", () => {
  test("normal selectors", async () => {
    const code = await generateCode("test/samples/normal_selectors.js");
    expect(code).toMatchSnapshot();
  });

  test("hover selectors", async () => {
    const code = await generateCode("test/samples/hover_selectors.js");
    expect(code).toMatchSnapshot();
  });

  test("media selectors", async () => {
    const code = await generateCode("test/samples/media_selectors.js");
    expect(code).toMatchSnapshot();
  });

  test("focus selectors", async () => {
    const code = await generateCode("test/samples/focus_selectors.js");
    expect(code).toMatchSnapshot();
  });
});
