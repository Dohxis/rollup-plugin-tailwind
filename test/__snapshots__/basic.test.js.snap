// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform tailwind focus selectors 1`] = `
"const css = {\\":focus\\":{\\"backgroundColor\\":\\"#38c172\\"}};
"
`;

exports[`transform tailwind hover selectors 1`] = `
"const css = {\\":hover\\":{\\"backgroundColor\\":\\"#38c172\\"}};
"
`;

exports[`transform tailwind media selectors 1`] = `
"const css = {\\"@media (min-width: 576px)\\":{\\"color\\":\\"#38c172\\"},\\"@media (min-width: 768px)\\":{\\"fontSize\\":\\"1.5rem\\"},\\"@media (min-width: 992px)\\":{\\"textTransform\\":\\"capitalize\\"}};
"
`;

exports[`transform tailwind normal selectors 1`] = `
"const css = {\\"color\\":\\"#fff\\",\\"fontWeight\\":\\"500\\",\\"fontSize\\":\\"1.5rem\\",\\"display\\":\\"flex\\",\\"textTransform\\":\\"capitalize\\"};
"
`;
